packages/portalnetwork/src/client/provider.ts

```diff
@@ -10,69 +10,68 @@ import {
 import { PortalNetwork } from './client.js'
 import { PortalNetworkOpts } from './types.js'
 
-export class UltralightProvider extends ethers.providers.StaticJsonRpcProvider {
-  private fallbackProvider:
-    | ethers.providers.StaticJsonRpcProvider
-    | ethers.providers.JsonRpcProvider
+export class UltralightProvider extends ethers.JsonRpcProvider {
+  private fallbackProvider: ethers.JsonRpcProvider
   public portal: PortalNetwork
   public historyProtocol: HistoryProtocol
   public static create = async (
-    fallbackProviderUrl: string | ethers.providers.JsonRpcProvider,
+    fallbackProviderUrl: string | ethers.JsonRpcProvider,
     network = 1,
     opts: Partial<PortalNetworkOpts>,
   ) => {
     const portal = await PortalNetwork.create(opts)
     return new UltralightProvider(fallbackProviderUrl, network, portal)
   }
   constructor(
-    fallbackProvider: string | ethers.providers.JsonRpcProvider,
+    fallbackProvider: string | ethers.JsonRpcProvider,
     network = 1,
     portal: PortalNetwork,
   ) {
     super(
-      typeof fallbackProvider === 'string' ? fallbackProvider : fallbackProvider.connection,
+      //@ts-ignore
+      typeof fallbackProvider === 'string' ? fallbackProvider : fallbackProvider.#connect.url,
```

Comment:
what is going on with `fallbackProvider.#connect.url`?

Reply:
```suggestion
      typeof fallbackProvider === 'string' ? fallbackProvider : fallbackProvider.connection.url,
```

Reply:
I dunno.  Still trying to work that out. The new and improved `ethers` is quite a bit different than v5.7 was.